numpy/f2py/capi_maps.py

```diff
@@ -342,9 +342,9 @@ def getstrlength(var):
 def getarrdims(a, var, verbose=0):
     ret = {}
     if isstring(var) and not isarray(var):
-        ret['dims'] = getstrlength(var)
-        ret['size'] = ret['dims']
-        ret['rank'] = '1'
+        ret['size'] = getstrlength(var)
+        ret['rank'] = '0'
+        ret['dims'] = ''
```

Comment:
As per the Fortran standard, `rank=0` is only for scalars, though setting `dims` to be the same as `size` is definitely buggy.

Reply:
As per the J3/18-007r1 interpretation document for Fortran 2018 we have:

![image](https://user-images.githubusercontent.com/4336207/218253880-b8cacd89-563a-4dfb-bcbf-be7c85a7f071.png)

![image](https://user-images.githubusercontent.com/4336207/218253885-0bcb364c-c1e1-438c-bed7-ed56fb7fbaba.png)

It would seem, that the rank should remain `1`. The problem seems to be that the `size` parameter does not correspond to the Fortran standard and is instead applied to each element of the array, as noted in #23192.
```
{'dims': '8', 'size': '8', 'rank': '1'}
```
Is incorrect, but should be `dims: 1`, not `rank: 0`.

Reply:
They way I coded it is consistent with the other sections.   The string is intended to ge a scalar.  

Irrespective the Fortran nomenclature, what matters is what goes as definitions in the C code created; this is what affects what NumPy / Python interface sees.   This is the part that was wrong, there was no change to Fortran codes generated.

Reply:
So, I disagree, the rank should be 0, as we do want a string scalar.  Consider the fotrtran code
```fortran
program test

  character(len=8) :: s
  character(len=12), dimension(5,7) :: a

  print*,'ranks', rank(a), rank(s)
  print*,'dims s'
  print*,'dims a', shape(a)

end program test
```
gives
```
 ranks           2           0
 dims s
 dims a           5           7
```

Reply:
Maybe there was a misunderstanding by @HaoZeke about the FOrtran standard or the purpose of this fix.

What was coded is exactly what it should be.  Otherwise it will continue to be broken.

Reply:
Sorry, I was misreading the bug. Yes, for `isstring(var) and not isarray(var)` these parameters are correct. Thanks!